components/locale/src/fallback/algorithms.rs

```diff
@@ -229,101 +175,23 @@ mod tests {
-            requires_data: false,
-            extension_key: None,
-            fallback_supplement: None,
-            expected_language_chain: &["en-u-sd-usca", "en"],
-            expected_region_chain: &["en-u-sd-usca", "en", "und-u-sd-usca"],
-        },
-        TestCase {
-            input: "en-US-u-hc-h12-sd-usca",
-            requires_data: false,
-            extension_key: None,
-            fallback_supplement: None,
-            expected_language_chain: &["en-US-u-sd-usca", "en-US", "en-u-sd-usca", "en"],
-            expected_region_chain: &["en-US-u-sd-usca", "en-US", "und-US-u-sd-usca", "und-US"],
-        },
-        TestCase {
-            input: "en-US-fonipa-u-hc-h12-sd-usca",
-            requires_data: false,
-            extension_key: Some(key!("hc")),
-            fallback_supplement: None,
-            expected_language_chain: &[
-                "en-US-fonipa-u-hc-h12-sd-usca",
-                "en-US-fonipa-u-sd-usca",
-                "en-US-fonipa",
-                "en-US",
-                "en-fonipa-u-hc-h12-sd-usca",
-                "en-fonipa-u-sd-usca",
-                "en-fonipa",
-                "en",
-            ],
-            expected_region_chain: &[
-                "en-US-fonipa-u-hc-h12-sd-usca",
-                "en-US-fonipa-u-sd-usca",
-                "en-US-fonipa",
-                "en-US",
-                "und-US-fonipa-u-hc-h12-sd-usca",
-                "und-US-fonipa-u-sd-usca",
-                "und-US-fonipa",
-                "und-US",
-            ],
-        },
-        TestCase {
-            input: "en-u-hc-h12-sd-usca",
-            requires_data: true,
-            extension_key: None,
-            fallback_supplement: None,
-            expected_language_chain: &["en-u-sd-usca", "en"],
-            expected_region_chain: &["en-US-u-sd-usca", "en-US", "und-US-u-sd-usca", "und-US"],
-        },
-        TestCase {
-            input: "en-Latn-u-sd-usca",
-            requires_data: true,
-            extension_key: None,
-            fallback_supplement: None,
-            expected_language_chain: &["en-u-sd-usca", "en"],
-            expected_region_chain: &["en-US-u-sd-usca", "en-US", "und-US-u-sd-usca", "und-US"],
-        },
-        TestCase {
-            input: "en-Latn-US-u-sd-usca",
-            requires_data: true,
-            extension_key: None,
-            fallback_supplement: None,
-            expected_language_chain: &["en-US-u-sd-usca", "en-US", "en-u-sd-usca", "en"],
-            expected_region_chain: &["en-US-u-sd-usca", "en-US", "und-US-u-sd-usca", "und-US"],
-        },
-        TestCase {
-            // TODO(#4413): -u-rg is not yet supported; when it is, this test should be updated
-            input: "en-u-rg-gbxxxx",
-            requires_data: false,
-            extension_key: None,
-            fallback_supplement: None,
-            expected_language_chain: &["en"],
-            expected_region_chain: &["en"],
-        },
```

Comment:
Issue: We should fix this in 2.0 because it might require an architecture change in LocaleFallabcker. https://github.com/unicode-org/icu4x/issues/4413